fx quant: enable linear-bn1d fusion for PTQ

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66484

Summary:

https://github.com/pytorch/pytorch/pull/50748 added linear - bn1d fusion
in Eager mode, for PTQ only. This PR also enables this in FX graph mode.

We reuse the existing conv-bn-relu fusion handler, renaming `conv` to
`conv_or_linear` for readability.

The QAT version is saved for a future PR, for both eager and FX graph.

Test Plan:

```
python test/test_quantization.py TestFuseFx.test_fuse_linear_bn_eval
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31575392](https://our.internmc.facebook.com/intern/diff/D31575392)

vkuzo committed:
fx quant: enable linear-bn1d fusion for PTQ

Summary:

https://github.com/pytorch/pytorch/pull/50748 added linear - bn1d fusion
in Eager mode, for PTQ only. This PR also enables this in FX graph mode.

We reuse the existing conv-bn-relu fusion handler, renaming `conv` to
`conv_or_linear` for readability.

The QAT version is saved for a future PR, for both eager and FX graph.

Test Plan:

```
python test/test_quantization.py TestFuseFx.test_fuse_linear_bn_eval
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
fix nit on "fx quant: enable linear-bn1d fusion for PTQ"


Summary:

https://github.com/pytorch/pytorch/pull/50748 added linear - bn1d fusion
in Eager mode, for PTQ only. This PR also enables this in FX graph mode.

We reuse the existing conv-bn-relu fusion handler, renaming `conv` to
`conv_or_linear` for readability.

The QAT version is saved for a future PR, for both eager and FX graph.

Test Plan:

```
python test/test_quantization.py TestFuseFx.test_fuse_linear_bn_eval
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31575392](https://our.internmc.facebook.com/intern/diff/D31575392)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31575392).

vkuzo committed:
Update on "fx quant: enable linear-bn1d fusion for PTQ"


Summary:

https://github.com/pytorch/pytorch/pull/50748 added linear - bn1d fusion
in Eager mode, for PTQ only. This PR also enables this in FX graph mode.

We reuse the existing conv-bn-relu fusion handler, renaming `conv` to
`conv_or_linear` for readability.

The QAT version is saved for a future PR, for both eager and FX graph.

Test Plan:

```
python test/test_quantization.py TestFuseFx.test_fuse_linear_bn_eval
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31575392](https://our.internmc.facebook.com/intern/diff/D31575392)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31575392).

jerryzh168 reviewed (approved):
so we only need to support this in fusion and not in quantization?

vkuzo committed:
Update on "fx quant: enable linear-bn1d fusion for PTQ"


Summary:

https://github.com/pytorch/pytorch/pull/50748 added linear - bn1d fusion
in Eager mode, for PTQ only. This PR also enables this in FX graph mode.

We reuse the existing conv-bn-relu fusion handler, renaming `conv` to
`conv_or_linear` for readability.

The QAT version is saved for a future PR, for both eager and FX graph.

Test Plan:

```
python test/test_quantization.py TestFuseFx.test_fuse_linear_bn_eval
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31575392](https://our.internmc.facebook.com/intern/diff/D31575392)

[ghstack-poisoned]
